clang/include/clang/Basic/AttrDocs.td

```diff
@@ -455,6 +455,64 @@ The SYCL kernel in the previous code sample meets these expectations.
   }];
 }
 
+def SYCLKernelEntryPointDocs : Documentation {
+  let Category = DocCatFunction;
+  let Content = [{
+The ``sycl_kernel_entry_point`` attribute specifies that a function definition
```

Comment:
This first sentence is... very difficult to understand/self referential. Reword please.

Reply:
Yeah. I rewrote a bunch of the doc. Hopefully better now.